### ISSUE TITLE
[no-unused-vars] false positive after upgrade to ^5.5.0

### ISSUE THREAD

**Issue by xboy2012** [typescript-eslint/typescript-eslint#4259]
![image](https://user-images.githubusercontent.com/7540144/144813794-e2d74583-bf44-45aa-941e-9deda50e1a63.png)

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/main/docs/linting/TROUBLESHOOTING.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": ["error", { "ignoreRestSiblings": true }]
  }
}
```

```TS
// your repro code case
import { Field, ObjectType } from '@nestjs/graphql';
import { Column, Entity } from 'typeorm';
import { BaseEntity } from '@lib';

@Entity()
@ObjectType({ description: 'User Entity' })
export class User extends BaseEntity {
  @Column({ comment: 'Name' })
  @Field({ description: 'Name' })
  public name: string;

  @Column({ comment: 'Nick' })
  @Field({ description: 'Nick' })
  public nickname?: string;
}

```

**Expected Result**

No errors.

**Actual Result**

'Field' is defined but never used.

'Column' is defined but never used.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.5.0` |
| `@typescript-eslint/parser`        | `5.5.0` |
| `TypeScript`                       | `4.5.2` |
| `ESLint`                           | `8.4.0` |
| `node`                             | `16.13.0` |

**Comment by bradzacher** [comment]
I am unable to repro this:
https://typescript-eslint.io/play/#ts=4.5.2&sourceType=module&code=JYWwDg9gTgLgBAbzgMWAUwDYBMA0cDyARgFZoDGMAKgJ5hpwC+cAZlBCHAOQACAdmgGcYxAQHoA5lACGYABYBHDJwDcAKFCRYiOAGEIGAK4heeAKK8YwGNUYs2HTtbrQQK1au7nL1gBQBKDyJSCho6HyQsQTIoYDBLCF4ALi4AVQE0KDgvK2pORgC0AA9NeDIMKQEBODSMxFU4OG49Q2NwuDJ2EDQLZM4AOSkuvIYAhu5UTCw2yIFo2PikrgGh-Pq4MANCDGAyOF5BtGShGN5xNTWm-SNeNo6QLp6lnYBrYdHGiexpqJi44ATen0Xm81hstjs9i99l0APxHGAnM6qBhAA&rules=N4IgAgLgngDgpgZwMYCcCWMIFpEBs0B2EA9AQPZYCuBlCcAJlgG4CGKCIAXANohwooyKEABpQaAObkUcAEqIIAZTQAjfAQkdOEFJTgBfALr6gA&tsConfig=N4IgpgHgDmBOCWBbMA7ALgQwDYBEwGMB7WDNYgZxAC41YBXMAXyA

**Comment by xboy2012** [comment]
sorry, after a detailed investigation. I found that there's another package using ’@typescript/eslint-parser‘ 4.x

The false positives might be caused by mixing of 4.x and 5.x.

I upgraded the related packages and ensure there's only 5.x version `@typescript-eslint`, everything is ok.

**Comment by xboy2012** [comment]
As a recommendation, In the docs, we can tell users to see if there are multiple version of `@typescript-eslint` which might cause problems
